tests/js/spec/views/stream/searchBar.spec.jsx

```diff
@@ -35,6 +35,25 @@ describe('SearchBar', function() {
     sandbox.restore();
   });
 
+  describe('componentWillReceiveProps()', function() {
+    it('should update state.query if props.query is updated from outside', function() {
+      let searchBar = shallow(<SearchBar query="one" />, options);
+
+      searchBar.setProps({query: 'two'});
+
+      expect(searchBar.state().query).toEqual('two');
+    });
+
+    it('should not reset user input if a noop props change happens', function() {
+      let searchBar = shallow(<SearchBar query="one" />, options);
+      searchBar.setState({query: 'two'});
+
+      searchBar.setProps({query: 'one'});
+
+      expect(searchBar.state().query).toEqual('two');
```

Comment:
Awesome, thanks for the tests!

Can we add an additional test similar to this one where after we set state to `query: 'two'`, we set props to `query: 'three'`?

Reply:
force-pushed to add
```
diff --git a/tests/js/spec/views/stream/searchBar.spec.jsx b/tests/js/spec/views/stream/searchBar.spec.jsx
index 841091566..41e262d1e 100644
--- a/tests/js/spec/views/stream/searchBar.spec.jsx
+++ b/tests/js/spec/views/stream/searchBar.spec.jsx
@@ -52,6 +52,15 @@ describe('SearchBar', function() {
 
       expect(searchBar.state().query).toEqual('two');
     });
+
+    it('should reset user input if a meaningful props change happens', function() {
+      let searchBar = shallow(<SearchBar query="one" />, options);
+      searchBar.setState({query: 'two'});
+
+      searchBar.setProps({query: 'three'});
+
+      expect(searchBar.state().query).toEqual('three');
+    });
   });
 
   describe('getQueryTerms()', function() {
```